contracts/InterchainTokenFactory.sol

```diff
@@ -215,6 +215,49 @@ contract InterchainTokenFactory is IInterchainTokenFactory, ITokenManagerType, M
         tokenId = _deployInterchainToken(salt, destinationChain, tokenName, tokenSymbol, tokenDecimals, minter_, gasValue);
     }
 
+    /**
+     * @notice The deployRemoteInterchainToken with originalChainName allows deploying a remote interchain token back to the same chain, which could cause issues with ITS Hub balance tracking.
+     * To fix this, the deployRemoteInterchainToken function was overloaded to prevent self-deployemnt while still allowing self-transfers.
```

Comment:
```suggestion
     * @notice Deploys a remote interchain token on a specified destination chain.
```
Should document what the method does. This method doesn't need to have any discussions about originalChain. It should be discussed in the other `deployRemoteInterchainToken` (and as a `@dev` comment). Put this method first, so users will see it first and use this by default.

This part of the comment `... which could cause issues with ITS Hub balance tracking ...` is vague and so doesn't make much sense to external users. It's just an internal note, it would be better to mention the larger point, something like: `originalChainName is only allowed to be '', i.e the current chain. Other source chains are not supported anymore to simplify ITS deployment behaviour.`

Reply:
Thanks for comment. updated.